app/assets/javascripts/omni-nav-v2.js

```diff
@@ -1,12 +1,59 @@
+
+
+var OffCanvasNav = (function() {
+
+  // Module Vars
+  var off_canvas_links;
+  var resizeId;
+
+  // Module Functions
+  var initialize = function() {
+    off_canvas_links = $('#js-off-canvas-nav > ul > li > a');
+    syncLinkWidths();
+
+    $('#js-off-canvas-trigger, #js-close-off-canvas-nav, #js-off-canvas-overlay').on('click', function(event) {
+      event.preventDefault();
+      $('#js-off-canvas-nav-container').toggleClass('open');
+      $('#js-off-canvas-overlay').toggleClass('active');
+      $('body').toggleClass('no-scroll');
+    });
+
+    $('#js-off-canvas-nav-container .toggle').on('click', function() {
+      $(this).parent().toggleClass('open'); // Targets li
+      $(this).parent().find('ul').slideToggle();
+    });
+
+    $(window).on('resize', function() {
+      clearTimeout(resizeId);
+      resizeId = setTimeout(syncLinkWidths, 500);
+    });
+  };
+
+  var syncLinkWidths = function() {
+    var width = $('#js-off-canvas-nav > ul').width();
+    off_canvas_links.css('width', width);
+  };
+
+  return {
+    init: initialize
+  };
+})();
```

Comment:
It might make more sense to put this under OmniNav2. It would never be used outside that context, would it? Even it did, it could be exposed through OmniNav2's public API.